### PR TITLE
feat(tests): add enoent tests

### PR DESCRIPTION
**Pull request by musikid**

### CODE REVIEW

File: rust/src/tests/link.rs [hunk 1]
```diff
@@ -140,3 +143,18 @@ fn unchanged_ctime_fails(ctx: &mut SerializedTestContext, ft: FileType) {
...
+// link/09.t
+crate::test_case! {
+    /// link returns ENOENT if the source file does not exist
+    enoent_source_not_exists
+}
+fn enoent_source_not_exists(ctx: &mut TestContext) {
+    let source = ctx.gen_path();
+    let dest = ctx.gen_path();
+
+    assert_eq!(link(&source, &dest), Err(Errno::ENOENT));
```
Comment (asomers) [inline rust/src/tests/link.rs:159]: Isn't this redundant with the first half of the "either" test case?

File: rust/src/tests/errors/enoent.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        assert_enoent_symlink!($syscall, |_ctx: &mut crate::runner::context::TestContext,
+                                             path: &std::path::Path| {
+                $syscall($( $($($before),* ,)? )? path $( $(, $($after),*)? )?)
+        });
+    };
+}
+
+pub(crate) use assert_enoent_symlink;
+
+/// Asserts that it returns ENOENT if the named file does not exist
```
Comment (asomers) [inline rust/src/tests/errors/enoent.rs:69]: The doc string looks copy/pasted.  Could you please fix it?

File: rust/src/tests/truncate.rs [hunk 3]
```diff
@@ -79,3 +81,7 @@ fn unchanged_ctime_failed(ctx: &mut SerializedTestContext) {
         });
     });
 }
+
+// (f)truncate/04.t
```
Comment (asomers) [inline rust/src/tests/truncate.rs:85]: Should there be other tests in ftruncate.rs?

**Review by asomers — CHANGES_REQUESTED** [review]

File: rust/src/tests/truncate.rs (hunk 3 above)
Comment (musikid) [reply rust/src/tests/truncate.rs:85]: The original suite tested only for truncate, even in ftruncate files. Should I add ftruncate too?

Comment (asomers) [reply rust/src/tests/truncate.rs:85]: What about files like tests/ftruncate/00.t ?  Doesn't that test ftruncate?

Comment (musikid) [reply rust/src/tests/truncate.rs:85]: Yes, it seems to be the only one with the 13.t. I'm not sure if it's supposed to be like this, or just a copy/paste error.

Comment (asomers) [reply rust/src/tests/truncate.rs:85]: Oh, I see now.  yeah, that definitely looks like a copy/paste bug in the original.

Comment (musikid) [reply rust/src/tests/truncate.rs:85]: Alright, I will add the `ftruncate` tests then!

Comment (musikid) [reply rust/src/tests/truncate.rs:85]: I think that I understand now why there wasn't tests for ftruncate. I don't think that it's possible to create files if the path is invalid...

File: rust/src/tests/link.rs (hunk 1 above)
Comment (musikid) [reply rust/src/tests/link.rs:159]: There were two distinct cases in the original. I don't think they're exactly the same, but I could be wrong.

**Review by asomers — APPROVED** [review]
LGTM, though you'll have to resolve the conflict.

**Pull request merged**